components/AutocompleteSelect/index.js

```diff
@@ -201,6 +215,20 @@ class AutoCompleteSelect extends HTMLElement {
       existingList.remove();
     }
     const ul = document.createElement('ul');
+    ul.addEventListener('click', e => {
```

Comment:
@giladt hey, i've changed it to event delegation, so that we dont create a new handler on every list item, especially that there can thousands, as you said
lemme know what you think